src/Service/CaseManager.php

```diff
@@ -198,65 +196,98 @@ public function handleOS2FormsCaseSubmission(string $sender, array $submissionDa
 
     public function deleteCase(CaseEntity $case)
     {
-        foreach ($case->getCasePartyRelation() as $relation) {
-            // Check if the party is only related to this case and remove it if so.
-            $party = $relation->getParty();
-            $partyRelations = $this->entityManager
-                ->getRepository(CasePartyRelation::class)
-                ->findBy(['party' => $party])
+        // Remote notes
+        $notes = $case->getNotes();
+
+        foreach ($notes as $note) {
+            $this->entityManager->remove($note);
+        }
+
+        // Remove Agenda stuff - but keep party till later.
+        $agendaCaseItems = $case->getAgendaCaseItems();
+
+        foreach ($agendaCaseItems as $agendaCaseItem) {
+            /** @var DigitalPost[] $digitalPosts */
+            $digitalPosts = $this->entityManager
+                ->getRepository(DigitalPost::class)
+                ->findBy(['entityId' => $agendaCaseItem])
```

Comment:
Consider injecting the `DigitalPostRepository` instead of getting it here (and below). This will also help the static type checker (cf. https://github.com/itk-dev/naevnssekretariatet/blob/develop/src/Repository/DigitalPostRepository.php#L15).

Reply:
Updated!